app/src/main/java/com/practicum/playlistmaker/SearchHistory.kt

```diff
@@ -0,0 +1,51 @@
+package com.practicum.playlistmaker
+
+import android.content.SharedPreferences
+import com.google.gson.Gson
+
+class SearchHistory(private val tracksInHistory: ArrayList<Track>) {
+
+    fun getTracksFromSharedPrefs(sharedPrefs: SharedPreferences): ArrayList<Track> {
+
+        val historyInSharedPrefs = sharedPrefs.getString(HISTORY_SEARCH_KEY, null)
+
+        if(historyInSharedPrefs != null) {
+            val tracksFromSharedPrefs = Gson().fromJson(historyInSharedPrefs, Array<Track>::class.java)
+            for(track in tracksFromSharedPrefs) {
+                tracksInHistory.add(track)
+            }
```

Comment:
🍏 Можно сразу добавить весь список:
```suggestion
            tracksInHistory.addAll(tracksFromSharedPrefs)
```